pkg/fftm/manager.go

```diff
@@ -90,13 +91,14 @@ func NewManager(ctx context.Context, connector ffcapi.API) (Manager, error) {
 
 func newManager(ctx context.Context, connector ffcapi.API) *manager {
 	m := &manager{
-		connector:         connector,
-		apiServerDone:     make(chan error),
-		metricsServerDone: make(chan error),
-		metricsEnabled:    config.GetBool(tmconfig.MetricsEnabled),
-		eventStreams:      make(map[fftypes.UUID]events.Stream),
-		streamsByName:     make(map[string]*fftypes.UUID),
-		metricsManager:    metrics.NewMetricsManager(ctx),
+		connector:                connector,
+		apiServerDone:            make(chan error),
+		monitoringServerDone:     make(chan error),
+		deprecatedMetricsEnabled: config.GetBool(tmconfig.DeprecatedMetricsEnabled),
+		monitoringEnabled:        config.GetBool(tmconfig.MonitoringEnabled),
```

Comment:
just a thought: why do we need to store both booleans? Can we just store that it's enabled and store the config section that we use is either the deprecated one or the new one. Then the code doesn't need all `or` checks

Reply:
when creat an HTTP server, if deprecatedMetricsEnabled, then use configuration under `metrics` section to initialised the HTTP server, if monitoringEnabled, then use the configuration under `monitoring` section to initialised the HTTP server.

Reply:
and when adding the metrics path (configurable) if deprecatedMetricsEnabled, then use the value of `metrics.path`, if monitoringEnabled, then use the value of `monitoring.metricsPath`.

Reply:
Okay so after doing some more digging I see where the complexity occurs because of the interface with the APIServer. To create an API Server, you need to pass in a `config.Section` directly and it's easier in the case to just check is it one or the other and then pass that through. I was thinking we could store the `config.Section` as a variable in the manager and then just use that and we don't need all the switching inside the manager and just at the top when reading the config.

Ideally in ff-common, we should have a constructor which can accept a structure as config so we can handle breaking changes by reading in all the config in each component and then converting to that structure. We have done this in the past